### ISSUE TITLE
object_detection: Pre-trained weights don't result in the reported mAP for resnet50 fpn

### ISSUE THREAD

**Issue by ghost** [tensorflow/models#5540]
------------------------

### System information
- **What is the top-level directory of the model you are using**:
`research/object_detection`
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Source
- **TensorFlow version (use command below)**:
tf.VERSION = 1.9.0-rc0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
cuda 9.2
- **GPU model and memory**:
1080 GTX, 10 GB
- **Exact command to reproduce**:

```
checkpoint_dir=model_data/coco_ckpt/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/
pipeline_config_path=samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config
python model_main.py \
    --checkpoint_dir=${checkpoint_dir} \
    --pipeline_config_path=${pipeline_config_path} \
    --run_once
```
The data was downloaded using the provided script and the tf records for COCO were generated using the provided config was changed to include this line:
    input_path: "tf_record_data/coco_val.tfrecord-*-of-00010"

### Describe the problem

The result from running the code gives 34.4 mAP, which is 0.6 mAP less than the reported mAP of 35.2 [This is reported in the comments in the config file:

```
# Achieves 35.2 mAP on COCO14 minival dataset. Doubling the number of training
# steps to 50k gets 36.9 mAP
```

My understanding from the COCO website is taht COCO14 minival = COCO17 eval set, which is consistent with the script used for downloading the coco data in this repository. 

```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.344
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.514
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.376
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.112
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.317
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.504
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.303
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.474
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.509
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.222
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.507
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.676
```

If you could help explain this discrepancy, that would be great!

A potentially related issue:

As far as I can tell, the COCO validation set has 5000 examples, not 8000 examples (e.g. the official coco website: http://cocodataset.org/#download, downloading the data myself).  The configs such as:
https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config#L187
have num_examples = 8000. Why is this?  [The evaluator throws out the repeat evaluations of the model, so it doesn't cause any major problems.]

As an aside, the config is misleading because if you do: input_path: "PATH_TO_BE_CONFIGURED/mscoco_val.record-00000-of-00010"
and just replace PATH_TO_BE_CONFIGURED, then you load only one shard (i.e. one tenth of the data).

**Comment by roadcode** [comment]
hi，do you use the imagenet pretrain model for training? i also want to reproduce the result

**Comment by aaa-github** [comment]
I am using the weights for the fully trained network which are available in the object detection model zoo: https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md   These weights override the backbone weights.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by jvishnuvardhan** [comment]
This is a stale issue. Please check with recent TF versions of models and open a new issue if this is still an issue with new models. Thanks!
